### ISSUE TITLE
[Design] Onsite Notifications: Notification documentation discovery 

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#62744]
## Background

- [Related Slack discussion](https://dsva.slack.com/archives/C909ZG2BB/p1690469834190609)
- [Additional document on expectations for this work](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/onsite-notifications/2023-scaling-onsite-notifications/notification-documentation-discovery.md)

As I've been thinking about the VANotify strategy re: onsite notifications, I've realized how overly focused we've been on the approach to notifications on [VA.gov](http://va.gov/). Written notifications aren't the only way we can draw people's attention to things. 

Additionally, we should be thinking about notifications in the larger context of [VA.gov](http://va.gov/) and its various types of alerts -- both personalized and not personalized. We may not be responsible as a team for things like system alerts, but any type of alert/notification -- personalized or not -- effects the overall user experience.

So, what I was thinking might be useful would be to take a sprint to document the following:

1. Generally speaking, the different types of alerts/notifications that typically appear on websites/products -- this would be things like system alerts, general updates, personalized notifications, alert indicators, tags, etc.
2. Document how we are or are not already handling these things on [VA.gov](http://va.gov/)

## Tasks

- [x] Review [expectations document](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/onsite-notifications/2023-scaling-onsite-notifications/notification-documentation-discovery.md) and let Samara know if there are any questions
- [x] Begin research/documentation
- [x] 2 -3 days into doing this work, have a short regroup with Samara to make sure we're on the right track
- [ ] Complete research/documentation
- [ ] Share documentation with Samara/team

## Acceptance criteria

- [ ] Documentation is complete and added to [this Github folder](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/identity-personalization/onsite-notifications/2023-scaling-onsite-notifications)

**Comment by Samara-Strauss** [comment]
Tagging @ajakabcin and @mtcA6 for awareness. Feel free to work with the designers to figure out how to assign this work.

**Comment by pamela-drouin** [comment]
@andaleliz @aagosto90 A great first place to mine would be OS design systems like Apple or Android, and then other public design systems like Shopify, etc. 😄

**Comment by Samara-Strauss** [comment]
[My VA 2023 Notification Documentation Discovery](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/onsite-notifications/2023-scaling-onsite-notifications/notification-documentation-discovery-findings.md)

Great job, @aagosto90!! Thank you again for doing this work.

To reiterate what we spoke about in the meeting this morning:

- Can you please add documentation on the error alert (ie. the red alert) and how that is different from the warning (yellow) alert?
- I am curious if there are other examples besides the new claim card design where people are using the small yellow alert state for personalized action items and what your opinion is on how we should move forward from here. 
- (And a thing we didn't talk about) For the intercept modal, I think there are some instances of this on VA.gov. I know there is a customer survey that pops up occasionally, and we have one in the profile for when someone changes their home address and it's different than their mailing address. Anyway, since the modal is classified as something we're not using, I wanted to let you know there are a few instances where we are using intercept modals already. [Here's the design system doc on modals](https://design.va.gov/components/modal).

**Comment by andaleliz** [comment]
@aagosto90 this is awesome, you did a great job with this! It feels comprehensive to me, just have some thoughts to share:
- **Event-based notifications opportunities:** thinking about the "undo" action in the component you referenced makes me think about other actions a person could or might want to take through a notification like that. This isn't a great example but maybe there's some kind of a confirmation we need from the user where we just offer them a button to do it right there in the notification and not send them elsewhere to do it.  Also gets my gears going about links vs buttons, and how if we had an action available in a notification like that, we'd def need a tertiary or small button to distinguish it from a link.
- Snackbar is such a funny name for a component 😆 
- Needing something updated/reviewed could be a valid use case for a "takeover" intercept modal, if there was ever something critical.  The modal Samara is referring to is definitely an intercept modal but it's the normal size and not a full page take over like I've seen on other sites (and what I think is captured in at least some of your screenshots)

**Comment by aagosto90** [comment]
Update on this:
- Thank you so much for the feedback @andaleliz and @Samara-Strauss !!
- I've updated the [documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/onsite-notifications/2023-scaling-onsite-notifications/notification-documentation-discovery-findings.md#types-of-notifications-were-already-using-and-may-not-need-to-expand-upon) so that now the category of "Alerts" is more broad and explains how and when we use each of them.
- I updated the [takeover modal documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/onsite-notifications/2023-scaling-onsite-notifications/notification-documentation-discovery-findings.md#types-of-notifications-were-already-using-and-may-have-an-opportunity-to-expand-upon) to include some examples of how we use that on VA.gov and a link to the component. I still don't see a pressing need for this type of alert/notification but think Liz brings up a good use case example. Other teams with critical updates are treating them using alerts, which brings me to my next point...
- I'm [chatting with Skyler](https://dsva.slack.com/archives/C04KHCT3ZMY/p1692127548147549), who proposed the slim alert within a card as an experimental design because I wasn't finding much just poking around on the website. It seems like they're the first to explore adding those alerts to a card and personally I think it's a great idea and a concept we may want to explore for My VA! I'm trying to get a sense of the scope of that treatment (still have an open question to Skyler) so not quite ready to update the documentation yet but I think that's the last piece here.

**Comment by aagosto90** [comment]
Another update:

- I've put these into one table because in the end pretty much everything had an opportunity to improve or expand upon
- I've elaborated on the usage of alerts in the table to include when we use alerts to point users to important action items on the page or within a card and linked to my Slack convo with Skyler about how they plan to use the new proposed card style with a slim alert in it. I see an opportunity here to utilize that type of card elsewhere in the auth exp of VA.gov and noted that in the table.
- This project feels done to me and has produced lots of useful ideas for the future of auth exp on VA.gov! Let me know if we're okay to close this ticket @Samara-Strauss

**Comment by Samara-Strauss** [comment]
@aagosto90 this is great. A couple more things, and then yes, I think this is good to close out:

1. The blue/green alert style don't really seem like "notifications" to me. They seem more like informational callouts, which I wouldn't classify really as an alert or notification (ie. hey, there is a thing that has happened or is happening that you need to know). I think you can take those out, with the exception of how they are used for event-based notifications.
2. The slim alert that Skyler is using -- we should call out that the big opportunity there is to determine whether or not this style should be used for personalized alerts/action items or not. This style, IMO, is a "system" style that should be used for messaging that is not personalized (eg. this system is down, something went wrong, etc). Determining guidelines and providing that guidance is a big opportunity going forward.

**Comment by aagosto90** [comment]
Thanks for the quick feedback @Samara-Strauss ! I removed the blue and green alerts for the purpose of this exercise - makes sense to me. 

I agree that the [guidelines for alerts](https://design.va.gov/components/alert) in the VA design system are super fuzzy and that's caused confusion on our team and I'm sure others as well. (For instance, we're currently using the red alerts when a system is down but were told async they should be yellow, but that's not really clearly outlined in the design system documentation.) It seems like because of the ambiguous guidelines around alerts, teams are not using them in a consistent way. It was tough to find other examples of teams using alerts to draw attention to items needing attention as Skyler said no other team is using them in a card like this, but I remembered that we use that type of alert to draw attention to addresses needing updating in profile too, so they're not the only team to attempt to use alerts in this way, technically. I think our team can lead the charge in creating guidelines for elevating action items nested within pages and that's a big opportunity for us! 

Let me know if we're good to close this ticket now. Really appreciate your feedback and guidance on this project!
